From 33dd77df6b2fca62ed1bf2d76ab42b72e03933ea Mon Sep 17 00:00:00 2001
From: Monk Liu <Monk.Liu@amd.com>
Date: Thu, 11 May 2017 13:59:15 +0800
Subject: [PATCH 1553/2094] drm/amdgpu:use job's list instead of check fence

because if the fence is really signaled, it could already
released so the fence pointer is a wild pointer, but if
we use job->base.node we are safe because job will not
be released untill amdgpu_job_timedout finished.

Signed-off-by: Monk Liu <Monk.Liu@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Avinash M N <avimn@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_device.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
index 85929dc..632d6b5 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_device.c
@@ -2710,9 +2710,9 @@ int amdgpu_sriov_gpu_reset(struct amdgpu_device *adev, struct amdgpu_job *job)
 		if (job && j != i)
 			continue;
 
-		/* here give the last chance to check if fence signaled
+		/* here give the last chance to check if job removed from mirror-list
 		 * since we already pay some time on kthread_park */
-		if (job && fence_is_signaled(&job->base.s_fence->finished)) {
+		if (job && list_empty(&job->base.node)) {
 			kthread_unpark(ring->sched.thread);
 			goto give_up_reset;
 		}
-- 
2.7.4

